Perlin.py

```diff
@@ -0,0 +1,139 @@
+import decimal
+import os
+import sys
+import xml.etree.ElementTree as ET
+from xml.etree import ElementTree
+
+import numpy as np
+
+import noise
+
+if 'SUMO_HOME' in os.environ:
+    tools = os.path.join(os.environ['SUMO_HOME'], 'tools')
+    sys.path.append(tools)
+else:
+    sys.exit("Please declare environment variable 'SUMO_HOME' to use sumolib")
+
+import sumolib
+
+
+def drange(x, y, jump):
+    while x < y:
+        yield float(x)
+        x += decimal.Decimal(jump)
+
+
+def test_perlin_noise():
+    # Print 2d simplex noise in from x, y in 0..1 with step 0.1
+    for x in drange(0, 1.01, 0.1):
+        for y in drange(0, 1.01, 0.1):
+            print(f"[{x:.2},{y:.2}] {noise.snoise2(x, y)}")
+
+
+def get_shape_of_edge_name(net: sumolib.net.Net, edge: str) -> list:
+    """
+    Return the shape of an edge in coordinate form; [(x_1,y_1), (x_2,y_2), ... , (x_n,y_n)].
+    :param net: the SUMO network
+    :param edge: the edge ID
+    :return: a tuple or list of tuples that define the shape of the edge
+    """
+    return net.getEdge(edge).getShape()
+
+
+def get_edge_pair_centroid(coords: list) -> (float, float):
+    """
+    Centroid of rectangle (edge_pair) = (width/2, height/2)
+    :param coords: [(x_1,y_1), (x_2,y_2), ... , (x_n,y_n)]
+    :return: Centroid of given shape
+    """
+
+    x_avg = np.mean([pos[0] for pos in coords])
+    y_avg = np.mean([pos[1] for pos in coords])
+    return x_avg, y_avg
+
+
+def scale_noise(noise: float) -> float:
+    """
+    The 'noise' lib returns a value in the range of [-1:1]. The noise value is scaled to the range of [0:1].
+    :param noise: a float [-1:1]
+    :return: the noise value scaled to [0:1]
+    """
+    return (noise + 1) / 2
+
+
+def get_perlin_noise(x, y) -> float:
+    """
+    TODO: Find sane offset to combat zero-value at (0, 0)
+    :param x:
+    :param y:
+    :return:
+    """
+    return scale_noise(noise.snoise2(x, y))
```

Comment:
Either rename to `get_simplex_noise` or actually sample perlin noise instead. Also what about all the parameters like seed, scale, octaves, lucanarity, and persistence? I doubt all the default values are perfect for our case.

Reply:
Down the line it would be prudent to investigate what values of these parameters should take but at this point we don't know. I'll open an issue.

Also, it uses perlin noise now `pnoise`.